From c6cabb326d9af5164498b6b2d02e4f86c198fa85 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:27:34 -0400
Subject: [alsa] hda: Add missing loopback merge path for AD1884/1984 codecs

Message-id: <140972823573-227-patchset-jkysela@redhat.com>
Patchwork-id: 90927
O-Subject: [RHEL 7.1 PATCH 226/442] ALSA: hda - Add missing loopback merge path for AD1884/1984 codecs
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

The mixer widget (NID 0x20) of AD1884 and AD1984 codecs isn't
connected directly to the actual I/O paths but only via another mixer
widget (NID 0x21).  We need a similar fix as we did for AD1882.

Cc: <stable@vger.kernel.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: c5eda4c1bf6214332c46fb2f4e7c42a85e5e5643)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_analog.c b/sound/pci/hda/patch_analog.c
index 1eb00a6..fff09be 100644
--- a/sound/pci/hda/patch_analog.c
+++ b/sound/pci/hda/patch_analog.c
@@ -1091,6 +1091,7 @@ static int patch_ad1884(struct hda_codec *codec)
  spec = codec->spec;
 
  spec->gen.mixer_nid = 0x20;
+ spec->gen.mixer_merge_nid = 0x21;
  spec->gen.beep_nid = 0x10;
  set_beep_amp(spec, 0x10, 0, HDA_OUTPUT);
 
-- 
1.7.1